src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -666,7 +667,7 @@ public synchronized RejectedAccessException accessRejected(@Nonnull RejectedAcce
     }
 
     @Restricted(NoExternalUse.class) // implementation
-    @Extension public static final class ApprovedWhitelist extends ProxyWhitelist {
+    @Extension public static final class ApprovedWhitelist extends ProxyWhitelist implements CacheableWhitelist {
```

Comment:
No. This includes `AclAwareWhitelist`, remember.

Reply:
…and even if it did not, it would not be cacheable without some sort of reset flag, since its delegates are mutable. Basically no `ProxyWhitelist` should be a `CacheableWhitelist` IIUC.

Reply:
Oops, that's a legit bug -- that was not intentional and slipped through.